### ISSUE TITLE
Feature Request: Add 'Padding' Property to Network Graph Nodes

### ISSUE THREAD

**Issue by luxlogica** [almende/vis#1653]
Currently, when drawing network graphs, there is not a lot of spacing between a node's internal label and its border - the node's **internal padding** is minimal. Increasing this internal padding, however, does not seem to be a trivial matter. As a newbie, I've read through the docs, trawled through StackOverflow and Google, but have not found a way to do it.

Ideally, this should just be another node property that the user could set easily, like this:

``` javascript
var nodes = new vis.DataSet([
    {id: 1, label: 'parent', padding: 18},
    {id: 2, label: 'child 1', padding: 9},
    {id: 3, label: 'child 2', padding: 9}
]);
```

If there is an easy workaround that I've missed, I'd be extremely happy to find out!

**Comment by AlexDM0** [comment]
Hi,

This is not possible as an option, though adding spaces to the begin and end would effectively do the same. Since this is the canvas and not the DOM, padding as such does not exist.

Regards

**Comment by luxlogica** [comment]
@AlexDM0 thank you for the feedback. I had tried adding spaces before+after the string, but that alone does not increase padding above+below the string (and is hard to work out, and looks very odd, if there are multiple lines on the label).

I understand the limitations of the canvas. Off the top of my head, however, I can think of a couple of possible workarounds:

    1. If we were able to **scale** the node, *without scaling the internal label*, we'd effectively get the same result - i.e., the label could be scaled up x%, but the label remained the same size.
    2. If we were able to define more than one border for the node, we could effectively use a first border (with the same colour as the bg) as a 'padding', and a second border as an actual 'border'.

Right now, in order to get the padding I need, I'm using workaround #2 above: I'm using a border with the same colour as the background to get the 'extra padding' effect. Unfortunately, this means that my nodes cannot actually have a border...

**Comment by AlexDM0** [comment]
Hi,

It would help if you could provide an example of where it isn't sufficient. I'd rather not make an extra option for this. 

Regards

**Comment by luxlogica** [comment]
You can see a linked picture in the following StackOverflow discussion:

http://stackoverflow.com/questions/35374778/how-to-increase-label-padding-in-the-nodes-of-a-vis-js-network-graph

Unfortunately, the way the default **network graph** is drawn in Vis.js - i.e., look at the "Basic" example - does not conform to good design principles: a lot more *negative space* is needed around the labels in order for the nodes not to look cluttered and visually polluted. Negative space is an essential element of design, just as important as border, background or font. Not being able to configure it means that Vis.js is simply not able to be used in some projects - simply because its appearance cannot be sufficiently customised. 

Right now, I'm working on a project where I'll need another network graph, and vis.js would be wonderful to use. But in this project, the design calls for semi-transparent nodes - ie., the nodes will have to use an `rgba()` colour, with the alpha component set to `< 1`. This makes it impossible to use the workaround #2 above: to get the padding (= negative space) we need inside the nodes, we'd have to make the border semi-transparent as well. Doing so will make the semi-transparent border overlap with the semi-transparent background (border is drawn *centered* on the background edge), and we end up with no extra padding, and double border (an opaque 'internal' border and a semi-transparent 'external' one).

This alone has forced us to start looking for another graphing library - which I loathe to do, because Vis.js is so amazing, and its functionality absolutely perfect... :(

**Comment by AlexDM0** [comment]
We'll look into this. I cannot promise when this will be finished though. You're free to implement it yourself and do a pull request. If you're interested in helping out I can pinpoint where you need to add your changes.

Regards

**Comment by mojoaxel** [comment]
see also #1331

**Comment by mojoaxel** [comment]
fixed by @eymiha with #2343
